upgrade/tests/upgrade_package/conftest.py

```diff
@@ -0,0 +1,51 @@
+import os
+
+import pytest
+from mock import patch
+
+from upgrade.scripts.upgrade_python_package import pip
+from ..conftest import VENV_PATH, REPOSITORY_WHEELS_PATH
+
+CLOUDSMITH_URL = os.environ.get("CLOUDSMITH_URL", False)
+
+
+@pytest.fixture(autouse=True)
+def use_pip():
+    pip("install", "--upgrade", "pip")
+    yield pip
+    pip("uninstall", "-y", "oll-test-top-level")
+    pip("uninstall", "-y", "oll-dependency1")
+    pip("uninstall", "-y", "oll-dependency2")
+
+
+@pytest.fixture
+def constraints_dir():
+    return VENV_PATH / "lib" / "site-packages"
+
+
+@pytest.fixture()
+def mocked_constraints_path():
```

Comment:
Why is it necessary to mock constraints path and find spec?

Reply:
This `conftest.py` was copied over from the top level `conftest.py`, so it's the original testing framework that we did a while back ago, and couldn't remember why. I tried removing it but it seems like it was done to get CI (linux) tests working. After trying to get rid of them [in this commit](https://github.com/openlawlibrary/upgrade-python-package/pull/15/commits/db67dcaae49bd9552879d1b772ea37174213cbe1), the linux tests fail. I'm not sure what the difference here is on remote vs on local machine.

Reply:
Found the explanation:
* Without `find_spec` mocked, function won't find the installed module
  from the fresh venv, but from existing venv from which pytest was
  invoked. This lead to module not being run, even though if it's valid.
  Mock `find_spec` to result to True, and assert if the module was run
  from unit tests.

No reason to try spend any more time on that then